src/Json.php

```diff
@@ -84,6 +86,24 @@ public static function withParent(mixed $parent, callable $cb): mixed
         }
     }
 
+    public static function isInStrictMode(RClass $r): bool
+    {
+        $strictModeAttr = JsonStrictDeserialize::class;
```

Comment:
There's a lot of uncached reflection calls in here that could very easily be cached to avoid recomputing and re instantiating reflection classes on every property of a given class, and again on every instance of a given class.